lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -2918,9 +2918,9 @@ bool TypeChecker::typeCheckForEachBinding(DeclContext *dc, ForEachStmt *stmt) {
       SequenceType = solution.simplifyType(SequenceType);
 
       // Perform any necessary conversions of the sequence (e.g. [T]! -> [T]).
-      if (tc.convertToType(expr, SequenceType, cs.DC)) {
-        return nullptr;
-      }
+      expr = solution.coerceToType(expr, SequenceType, cs.getConstraintLocator(expr));
```

Comment:
What does this change? The affected test does not have a for..in loop

Reply:
@slavapestov You're right, this isn't related to the crash. But it is just a removal of an unnecessary call to `convertToType` since this method creates a new system(with a conversion constraint), solves it and only then perform the conversion. But at this point, there is already a solution that is being applied, so we just take the solution and let the rewriter perform the conversion using `coerceToType` on the solution that is already there.

Reply:
Ok, fair enough. I see this change is in a separate commit too